### PR TITLE
Spec for 3.28

### PR DESCRIPTION
**Pull request by hbons**
Notable changes:
- Convert spec to YAML
- Add dependencies for Ubuntu's AppIndicator
- Compile an optimised Git for bundling from source
- The build now includes the appdata file with screenshots
- Build with Meson

### CODE REVIEW

**Comment by hbons** [comment]
bot, build

**Comment by hbons** [comment]
bot, build

**Comment by nedrichards** [comment]
Could you add [OARS](https://hughsie.github.io/oars/) to the upstream appdata? Not a blocker but it's now a flathub requirement and we'd love to fixup all the apps that got in early.

**Review by nedrichards — APPROVED** [review]
other than the conituing sad state of mono gtk3 bindings, lgtm. It builds fine and I assume you've tested locally.

**Comment by hbons** [comment]
will do!

**Comment by hbons** [comment]
I've added this in master. Are the ratings needed right now? I don't want to create a new release just for that, but I can make a patch for the spec.

**Comment by nedrichards** [comment]
Not needed right now, happy to wait until the next release if that's sooner
than six months.

**Comment by hbons** [comment]
Ok! Definitely in the next six months. I'm syncing to the GNOME release schedule now (hence the version number change to match).

**Comment by hbons** [comment]
so is this good to be merged? :)

File: org.sparkleshare.SparkleShare.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                - sed 's/gtk_icon_info_free/g_object_unref/' -i libindicator/indicator-image-helper.c
+                - sed 's/LIBS="-lm  $LIBS"/LIBS=" -lm  $LIBS"/g' -i configure
+                - sed 's/LIBM="-lm"/LIBM=" -lm"/g' -i configure
+      config-opts:
+          - --with-gtk=3
+          - --disable-static
+          - --disable-tests
+      cleanup: ['/include', '/lib/*.la', '/lib/pkgconfig', '/libexec', '/share']
+
+    - name: libappindicator
```
Comment (TingPing) [inline org.sparkleshare.SparkleShare.yml]: You can replace this with the shared-module: https://github.com/flathub/shared-modules/blob/master/libappindicator/libappindicator-gtk3-12.10.json

Comment (hbons) [reply org.sparkleshare.SparkleShare.yml]: oh nice! let me try that.

**Comment by hbons** [comment]
bot, build

File: org.sparkleshare.SparkleShare.yml (hunk 1 above)
Comment (hbons) [reply org.sparkleshare.SparkleShare.yml]: changed, but the build bot seems to be having trouble with it... (it works for me locally)

Comment (TingPing) [reply org.sparkleshare.SparkleShare.yml]: Strange, there are probably half a dozen other flathub projects using it...

**Comment by hbons** [comment]
@nedrichards any clue why it doesn’t build?

**Comment by barthalion** [comment]
What if you skip the quotation marks?

**Comment by barthalion** [comment]
It's actually simpler, somehow submodule has not been added correctly, try to clone the repo with `--recursive --branch 3.28` to see it locally.

**Comment by nedrichards** [comment]
I'll check out the branch and have a look. It's absolutely something to do with the submodule.

**Comment by nedrichards** [comment]
bot, build

**Comment by hbons** [comment]
thanks!

**Pull request merged**